test/Interop/SwiftToCxx/enums/swift-enum-implementation.swift

```diff
@@ -0,0 +1,263 @@
+// CHECK-NEXT:     swift::_impl::implClassFor<S>::type::initializeWithTake(result._getOpaquePointer(), swift::_impl::implClassFor<S>::type::getOpaquePointer(*valCopy));
+// CHECK-NEXT:     result._destructiveInjectEnumTag(2);
+// CHECK-NEXT:     return result;
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline bool E::isZ() const {
+// CHECK-NEXT:     return *this == E::z;
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline S E::getZ() const {
+// CHECK-NEXT:     if (!isZ()) abort();
+// CHECK-NEXT:     alignas(E) unsigned char buffer[sizeof(E)];
+// CHECK-NEXT:     auto *thisCopy = new(buffer) E(*this);
+// CHECK-NEXT:     char * _Nonnull payloadFromDestruction = thisCopy->_destructiveProjectEnumData();
+// CHECK-NEXT:     return swift::_impl::implClassFor<S>::type::returnNewValue([&](char * _Nonnull result) {
+// CHECK-NEXT:       swift::_impl::implClassFor<S>::type::initializeWithTake(result, payloadFromDestruction);
+// CHECK-NEXT:     });
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline E E::_impl_w::operator()(swift::Int val) const {
+// CHECK-NEXT:     auto result = E::_make();
+// CHECK-NEXT:     memcpy(result._getOpaquePointer(), &val, sizeof(val));
+// CHECK-NEXT:     result._destructiveInjectEnumTag(3);
+// CHECK-NEXT:     return result;
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline bool E::isW() const {
+// CHECK-NEXT:     return *this == E::w;
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline swift::Int E::getW() const {
+// CHECK-NEXT:     if (!isW()) abort();
+// CHECK-NEXT:     alignas(E) unsigned char buffer[sizeof(E)];
+// CHECK-NEXT:     auto *thisCopy = new(buffer) E(*this);
+// CHECK-NEXT:     char * _Nonnull payloadFromDestruction = thisCopy->_destructiveProjectEnumData();
+// CHECK-NEXT:     swift::Int result;
+// CHECK-NEXT:     memcpy(&result, payloadFromDestruction, sizeof(result));
+// CHECK-NEXT:     return result;
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline E E::_impl_auto::operator()(void * _Nonnull val) const {
+// CHECK-NEXT:     auto result = E::_make();
+// CHECK-NEXT:     memcpy(result._getOpaquePointer(), &val, sizeof(val));
+// CHECK-NEXT:     result._destructiveInjectEnumTag(4);
+// CHECK-NEXT:     return result;
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline bool E::isAuto_() const {
+// CHECK-NEXT:     return *this == E::auto_;
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline void * _Nonnull E::getAuto_() const {
+// CHECK-NEXT:     if (!isAuto_()) abort();
+// CHECK-NEXT:     alignas(E) unsigned char buffer[sizeof(E)];
+// CHECK-NEXT:     auto *thisCopy = new(buffer) E(*this);
+// CHECK-NEXT:     char * _Nonnull payloadFromDestruction = thisCopy->_destructiveProjectEnumData();
+// CHECK-NEXT:     void * _Nonnull result;
+// CHECK-NEXT:     memcpy(&result, payloadFromDestruction, sizeof(result));
+// CHECK-NEXT:     return result;
+// CHECK-NEXT:   }
+// CHECK-NEXT:   inline E E::_impl_foobar::operator()() const {
+// CHECK-NEXT:     auto result = E::_make();
+// CHECK-NEXT:     result._destructiveInjectEnumTag(5);
```

Comment:
Can you also add a test for resilient enum where we call `_destructiveInjectEnumTag` using extern "C" tag values as well?